test/fake_cloud_firestore_test.dart

```diff
@@ -824,7 +839,7 @@ void main() {
     final foo = firestore.collection('users').doc('foo');
     await foo.set(<String, dynamic>{'name.firstName': 'OldValue Foo'});
     final bar = firestore.collection('users').doc('bar');
-    await foo.set(<String, dynamic>{'name.firstName': 'OldValue Bar'});
+    await bar.set(<String, dynamic>{'name.firstName': 'OldValue Bar'});
```

Comment:
This was a pre-existing bug. I caught it when making `update` throw exceptions on non-existent documents. On line 831 below, bar was actually still unset and `batch.update(bar` threw an exception.